### DISCUSSION TITLE
tutorials/intro-dgs/13-wrapping-up

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#151] [course feedback]
# tutorials/intro-dgs/13-wrapping-up

https://www.apollographql.com/tutorials/intro-dgs/13-wrapping-up

**Comment by rafaelo0321** [comment]
Es un excelente curso muchas gracias

**Comment by boolivar** [comment]
One thing to note here: mutation makes extra call to amenities through amenities resolver.
Depending on desired behavior this call can be removed using `DataFetcherResult` with local context:
```java
return DataFetcherResult.<CreateListingResponse>newResult()
    .data(response)
    .localContext(Map.of("hasAmenityData", true))
    .build();
```
Such behavior applicable in our case since create listings response contains full amenities data.

**Reply by AhmedRMunir** [reply]
Nice catch

**Comment by whitecat** [comment]
In the section `Running a simple mutation` the expected response is a little off. The response it includes the `id` in the listing and not the `title`. Maybe the area where the tutorial says `fill out the following operation:` should be updated to

```mutation CreateListing($input: CreateListingInput!) {
  createListing(input: $input) {
    code
    success
    message
    listing {
      id
      description
      costPerNight
      amenities {
        name
        category
      }
    }
  }
}
```

**Reply by AhmedRMunir** [reply]
bump
